juntagrico/views.py

```diff
@@ -67,27 +67,37 @@ def job(request, job_id):
     job_is_running = job.start_time() < timezone.now()
     job_canceled = job.canceled
     can_subscribe = job.infinite_slots or not (job_fully_booked or job_is_in_past or job_is_running or job_canceled)
+    can_unsubscribe = job.allow_unsubscribe and len(AssignmentDao.assignments_for_job_and_member(job_id, member)) > 0
 
     if request.method == 'POST' and can_subscribe:
-        num = int(request.POST.get('jobs'))
-        if 0 < num and (job.free_slots >= num or job.infinite_slots):
-            # adding participants
-            amount = 1
-            if Config.assignment_unit() == 'ENTITY':
-                amount = job.multiplier
-            elif Config.assignment_unit() == 'HOURS':
-                amount = job.multiplier * job.duration
-            for i in range(num):
-                assignment = Assignment.objects.create(
-                    member=member, job=job, amount=amount)
-            for extra in job.type.job_extras_set.all():
-                if request.POST.get('extra' + str(extra.extra_type.id)) == str(extra.extra_type.id):
-                    assignment.job_extras.add(extra)
-            assignment.save()
-            membernotification.job_signup(member.email, job)
+        if request.POST.get('subscribe'):
+            num = int(request.POST.get('jobs'))
+            if 0 < num and (job.free_slots >= num or job.infinite_slots):
+                # adding participants
+                amount = 1
+                if Config.assignment_unit() == 'ENTITY':
+                    amount = job.multiplier
+                elif Config.assignment_unit() == 'HOURS':
+                    amount = job.multiplier * job.duration
+                for i in range(num):
+                    assignment = Assignment.objects.create(
+                        member=member, job=job, amount=amount)
+                for extra in job.type.job_extras_set.all():
+                    if request.POST.get('extra' + str(extra.extra_type.id)) == str(extra.extra_type.id):
+                        assignment.job_extras.add(extra)
+                assignment.save()
+                membernotification.job_signup(member.email, job)
```

Comment:
I have added here the following two lines:
`job.allow_unsubscribe = True
job.save()`

and it seems to do the trick. Now the unsubscribe button appears next to the subscribe one.